### ISSUE TITLE
Protontricks Error: 'Could not find configured Proton installation!'

### ISSUE THREAD

**Issue by BobWazinski** [Matoking/protontricks#149]
**Describe the bug**
Can't run the following command to install Windows Media Player for a game:
`protontricks 21690 -q wmp11`
Also, tried the below gui command but I receive the same error:
`protontricks 21690 --gui`

**To Reproduce**
1. I followed the Flatpak installation guide on Github.
2. The above mentioned command fails and the following error is displayed:
`protontricks (ERROR): Could not find configured Proton installation!
protontricks (ERROR): Active Proton installation could not be found automatically.
Proton installation could not be found!`

Additionally, I tried to run the Protontricks GUI:
`protontricks --gui`
But I receive the following output:
`Found no games. You need to launch a game at least once before Protontricks can find it.`
I obviously ran the game before.

**Expected behavior**
WMP11 should have been installed and I should be able to enjoy the timeless masterpiece.

**System (please complete the following information):**
 - Distro: 
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
 - Protontricks installation method: Flatpak, as mentioned above.
 - Protontricks version: protontricks (1.8.2.dev0+geb897c6.d20220414)
 - Steam version: I am not participating in Beta, so it's the stable deb release (latest).

**Additional context**
Resident Evil 5 does not work OOTB. Windows Media Player is needed for Resident Evil 5 cut-scenes. I used to be able to run the necessary command previously on the old computer. Everything used to work. But again that was half a year ago, that was an old version of Pop!_OS. Don't remember from what repository I installed Protontricks previously.

**Comment by Matoking** [comment]
Could you run the command again using the `-v` flag using the following command:

`protontricks -v 21690 -q wmp11`

and copy the full output?

Since you've installed Protontricks using Flatpak, you'll need to grant additional filesystem permissions if you have Steam library folders outside the default location. Protontricks _should_ print a warning in the format "Steam library folder X not found. Protontricks might not have access to the directory." if that is the case, as well as a warning dialog if Protontricks is running in a Flatpak instructing how to change the permissions. The instructions for doing so can be found [here](https://github.com/flathub/com.github.Matoking.protontricks#configuration).

If no warning is printed at all that's still a problem, as Protontricks should detect that.

**Comment by BobWazinski** [comment]
Here's the output for `protontricks -v 21690 -q wmp11`:
protontricks (INFO): Running inside Flatpak sandbox, version 1.12.7.
protontricks (INFO): Found Steam directory at /home/bobwazinski/.var/app/com.valvesoftware.Steam/data/Steam. You can also define Steam directory manually using $STEAM_DIR
protontricks (INFO): Using default Steam Runtime at /home/bobwazinski/.var/app/com.valvesoftware.Steam/data/Steam/ubuntu12_32/steam-runtime
protontricks (INFO): WINETRICKS environment variable is not available. Searching from $PATH.
protontricks (INFO): Found 1 Steam library folders
protontricks (INFO): Currently logged-in Steam user: keanulover69
protontricks (INFO): Couldn't find custom shortcuts. Maybe none have been created yet?
protontricks (INFO): No compatibility tool found by reading Steam configuration. Using stable version of Proton as fallback.
protontricks (ERROR): Could not find configured Proton installation!
protontricks (ERROR): Active Proton installation could not be found automatically.
Proton installation could not be found!

**Comment by Matoking** [comment]
The line `protontricks (INFO): Found Steam directory at /home/bobwazinski/.var/app/com.valvesoftware.Steam/data/Steam. You can also define Steam directory manually using $STEAM_DIR` means that Protontricks detected the Flatpak installation of Steam and is prioritizing it, but you mentioned that you also installed the Debian version of Steam as well.

If both are installed, you'll need to choose the non-default installation using the `STEAM_DIR` environment variable like so:

```
$ STEAM_DIR=/home/bobwazinski/.steam/steam flatpak run com.github.Matoking.protontricks --gui
```
 
If that solves the issue, you can update your alias to include the `STEAM_DIR=/home/bobwazinski/.steam/steam` part before the actual Flatpak command.

**Comment by BobWazinski** [comment]
Alright, so running protontricks commands with manually defining the Steam directory helps. GUI detects installed games. Unfortunately, trying to install wmp11 results in another issue. 

The output after running the above suggested GUI command and choosing wmp11 install gives the following output:

`/home/bobwazinski/.var/app/com.github.Matoking.protontricks/cache/protontricks/proton/Proton 5.0/bin/bwrap-launcher: line 83: /run: Is a directory
/home/bobwazinski/.var/app/com.github.Matoking.protontricks/cache/protontricks/proton/Proton 5.0/bin/bwrap-launcher: line 83: exec: /run: cannot execute: Is a directory
Gtk-Message: 16:28:16.046: Failed to load module "appmenu-gtk-module"
Gtk-Message: 16:28:16.095: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:16.095: Failed to load module "canberra-gtk-module"
winetricks GUI enabled, using zenity 3.32.0
Executing mkdir -p /home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Gtk-Message: 16:28:19.920: Failed to load module "appmenu-gtk-module"
Gtk-Message: 16:28:19.968: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:19.968: Failed to load module "canberra-gtk-module"
Using winetricks 20220411 - sha256sum: 69856050cfe9d97902dfd68c1ad42a10a4be0fe76660744126d7e3ed75610495 with wine-5.0-655-g2409bd1 and WINEARCH=win64
Gtk-Message: 16:28:22.972: Failed to load module "appmenu-gtk-module"
Gtk-Message: 16:28:23.012: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:23.012: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:32.068: Failed to load module "appmenu-gtk-module"
Gtk-Message: 16:28:32.116: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:32.116: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:36.508: Failed to load module "appmenu-gtk-module"
Gtk-Message: 16:28:36.547: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:36.547: Failed to load module "canberra-gtk-module"
Executing w_do_call wmp11
Executing mkdir -p /home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Gtk-Message: 16:28:47.753: Failed to load module "appmenu-gtk-module"
Gtk-Message: 16:28:47.792: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:47.792: Failed to load module "canberra-gtk-module"
Executing load_wmp11 
Executing w_do_call wsh57
Executing mkdir -p /home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Gtk-Message: 16:28:51.838: Failed to load module "appmenu-gtk-module"
Gtk-Message: 16:28:51.883: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:51.883: Failed to load module "canberra-gtk-module"
Executing load_wsh57 
Executing cabextract -q -d /home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690/pfx/dosdevices/c:/windows/syswow64 /home/bobwazinski/.var/app/com.github.Matoking.protontricks/cache/winetricks/wsh57/scripten.exe
/home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690/pfx/dosdevices/c:/windows/syswow64/spmsg.dll: Permission denied
/home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690/pfx/dosdevices/c:/windows/syswow64/spuninst.exe: Permission denied
/home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690/pfx/dosdevices/c:/windows/syswow64/wscript.hlp: Permission denied
/home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690/pfx/dosdevices/c:/windows/syswow64/update/spcustom.dll: Permission denied
/home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690/pfx/dosdevices/c:/windows/syswow64/update/updspapi.dll: Permission denied
/home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690/pfx/dosdevices/c:/windows/syswow64/update/update.exe: Permission denied
------------------------------------------------------
warning: Note: command cabextract -q -d /home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690/pfx/dosdevices/c:/windows/syswow64 /home/bobwazinski/.var/app/com.github.Matoking.protontricks/cache/winetricks/wsh57/scripten.exe returned status 1. Aborting.
------------------------------------------------------
Gtk-Message: 16:28:54.043: Failed to load module "appmenu-gtk-module"
Gtk-Message: 16:28:54.084: Failed to load module "canberra-gtk-module"
Gtk-Message: 16:28:54.084: Failed to load module "canberra-gtk-module"`

Additional context: I have installed wmp11 previously using winetricks. So possibly the issue is related to winetricks.

**Comment by Matoking** [comment]
Yes, this is a known issue with Winetricks and Wine prefixes created by Proton.

The "Permission denied" error with cabextract makes this a duplicate of #97. The issue on Winetricks side is tracked at Winetricks/winetricks#1648.

**Comment by BobWazinski** [comment]
> Yes, this is a known issue with Winetricks and Wine prefixes created by Proton.
> 
> The "Permission denied" error with cabextract makes this a duplicate of #97. The issue on Winetricks side is tracked at [Winetricks/winetricks#1648](https://github.com/Winetricks/winetricks/issues/1648).

So yeah, I managed to find a workaround. Never had so many issues at once. I think originally protontricks struggled to automatically detect Steam directory for debian-installation because I once installed the flatpak package of Steam which I uninstalled later on and switched to debian. `/home/bobwazinski/.var/app/com.valvesoftware.Steam/data/Steam` is of course the Flatpak directory, something I learned just recently. I noticed other programs like ProtonUp-Qt set Flatpak directory automatically even though it is absent (or was not uninstalled properly).

I didn't manage to make protontricks functioning because of the above issue but my workaround is simply using winetricks:
`WINEPREFIX=/home/bobwazinski/.steam/debian-installation/steamapps/compatdata/21690/pfx winetricks wmp11`

I don't know what wsh57 has anything to do with wmp11 install but I followed the workaround (uncommenting a line in winetricks script) suggested at [https://github.com/Winetricks/winetricks/issues/1648](url). However, this step might be unnecessary and the above winetricks command should be enough.

I also removed the proton prefix through `rm -rfv ~/.steam/steam/steamapps/compatdata/21690` but this is probably unnecessary too.

I'm leaving it here only in case anyone else will run into the same loop of issues.

Thanks a lot for the help! Will pray to the Linux Gods for your well-being.
